cmd/client/cmd/check.go

```diff
@@ -47,24 +50,55 @@ func check(rootArgs *rootArgs, printf, fatalf shared.FormatFn) {
 		fatalf("%v", err)
 	}
 
-	var cs *clientState
-	if cs, err = createAPIClient(rootArgs.mixerAddress, rootArgs.enableTracing); err != nil {
-		fatalf("Unable to establish connection to %s", rootArgs.mixerAddress)
-	}
-	defer deleteAPIClient(cs)
-
-	span, ctx := ot.StartSpanFromContext(context.Background(), "mixc Check", ext.SpanKindRPCClient)
-	for i := 0; i < rootArgs.repeat; i++ {
-		request := mixerpb.CheckRequest{Attributes: *attrs}
-		response, err := cs.client.Check(ctx, &request)
-
-		if err == nil {
-			printf("Check RPC completed successfully. Check status was %s", decodeStatus(response.Status))
-			dumpAttributes(printf, fatalf, &response.Attributes)
-		} else {
-			printf("Check RPC failed with: %s", decodeError(err))
-		}
+	// TODO: one span for each request - not sure what we can get from the per-stream span.
+
+	t := time.Now()
+
+	var wg sync.WaitGroup
+	wg.Add(rootArgs.clients)
+
+	for s := 0; s < rootArgs.clients; s++ {
```

Comment:
This is sending 'repeat' x 'clients' number of requests which is not what is implied by the command-line args. It should only be sending 'repeat' requests overall, split between 'clients' senders.

Instead of using channels here, it'd be more efficient to use WaitGroups.